### PR TITLE
document optional packages

### PR DESCRIPTION
**Pull request by jrobinAV**
I added a section for core in the ref man. 
It includes the note on optional packages. I re-used the note from taipy-gui, but I slightly modified the form (found it better). @FabienLelaquais, you might want to change the one on Gui if you think it s better as well.

### CODE REVIEW

File: src/taipy/core/__init__.py [hunk 1]
```diff
@@ -46,3 +46,22 @@
     untag,
 )
 from .task.task import Task
+
+"""# Taipy Core.
+
+The Taipy Core package is a python library made to build powerful and customized data-driven back-end applications.
```
Comment (FabienLelaquais) [inline src/taipy/core/__init__.py]: "python" -> "Python"

File: src/taipy/core/__init__.py [hunk 2]
```diff
@@ -46,3 +46,22 @@
...
+It provides the necessary tools to help Python developer transform their algorithms into a complete
```
Comment (FabienLelaquais) [inline src/taipy/core/__init__.py]: developer -> developers

**Review by FabienLelaquais — DISMISSED** [review]
Two minor typos.

File: src/taipy/core/__init__.py (hunk 1 above)
Comment (jrobinAV) [inline src/taipy/core/__init__.py]:
```suggestion
The Taipy Core package is a Python library made to build powerful and customized data-driven back-end applications.
```

File: src/taipy/core/__init__.py (hunk 2 above)
Comment (jrobinAV) [inline src/taipy/core/__init__.py]:
```suggestion
It provides the necessary tools to help Python developers transform their algorithms into a complete
```

**Pull request merged**